### ISSUE TITLE
feature request : Handling images 

### ISSUE THREAD

**Issue by zifken** [shady2k/logseq-inbox-telegram-plugin#2]
This is a great project that I've been using for quite some time.
I think a way to improve the productivity of a user would be the possibility of adding image uploading as a block.
I don't know if it's even possible, but I often need to quickly send the picture of a receipt or a document that i want to deal with later. Although the _beta_ mobile app of Logseq is very good I find  the telegram now + Logseq later much smoother.
Thank you

**Comment by shady2k** [comment]
At this moment you can send url with picture as text. About saving picture as asset in Logseq, I'm not sure, that this action possible by Logseq plugin API, I will check.

**Comment by zifken** [comment]
Thank you for your response. Apparently it is possible according to  this [phonetonote plugin issue](https://github.com/phonetonote/phonetonote-logseq/issues/2). However an approach could be to simply get the image URL from telegram and use a workflow like : 
- get `assets` folder from logseq API
- save image with telegram URL + `fetch` and `fs` to the folder 
- insert link to the editor with image URL 
I would try but I can't seem to find the full logseq plugin API if it even exists.

**Comment by shady2k** [comment]
> * save image with telegram URL + `fetch` and `fs` to the folder
As I know, `fs` module cannot be used inside plugins. This operation denied by Logseq.

**Comment by bepolymathe** [comment]
I’am interesting for that too if it’s possible. For images or any files...

**Comment by shady2k** [comment]
Yes, I want this feature too. I will ask logseq developers about it.

**Comment by L0cutus** [comment]
It is a **plus** also for me thanks for this great plugin!

**Comment by shady2k** [comment]
I'm just created discussion on forum, please, add your voice.
https://discuss.logseq.com/t/feature-request-ability-to-write-assets-from-plugin-api/5047

**Comment by bepolymathe** [comment]
Done

**Comment by imenvn** [comment]
I have tried to use this by using Imgur.
What I did?
1. send image to telegram bot.
2. then plugin first get the url of telegram image from telegram api
3. then send this url to imgur via its api.
4. receiving Link to image from imgur.

But I can only able to get that link in Logs in console. I am not able to push that message with link to logseq.

Need Help with this regard.

**Comment by Fillll** [comment]
> I'm just created discussion on forum, please, add your voice. https://discuss.logseq.com/t/feature-request-ability-to-write-assets-from-plugin-api/5047

✅ Done.
